packages/context/src/__tests__/unit/context-config.unit.ts

```diff
@@ -0,0 +1,131 @@
+  beforeEach(createContext);
+
+  describe('configure()', () => {
+    it('configures options for a binding before it is bound', () => {
+      const bindingForConfig = ctx.configure('foo').to({x: 1});
+      expect(bindingForConfig.key).to.equal(
+        BindingKey.buildKeyForConfig('foo'),
+      );
+      expect(bindingForConfig.tagMap).to.eql({config: 'foo'});
+    });
+
+    it('configures options for a binding after it is bound', () => {
+      ctx.bind('foo').to('bar');
+      const bindingForConfig = ctx.configure('foo').to({x: 1});
+      expect(bindingForConfig.key).to.equal(
+        BindingKey.buildKeyForConfig('foo'),
+      );
+      expect(bindingForConfig.tagMap).to.eql({config: 'foo'});
+    });
+  });
+
+  describe('getConfig()', () => {
+    it('gets config for a binding', async () => {
+      ctx.configure('foo').toDynamicValue(() => Promise.resolve({x: 1}));
+      expect(await ctx.getConfig('foo')).to.eql({x: 1});
+    });
+
+    it('gets local config for a binding', async () => {
+      ctx
+        .configure('foo')
+        .toDynamicValue(() => Promise.resolve({a: {x: 0, y: 0}}));
+      ctx.configure('foo.a').toDynamicValue(() => Promise.resolve({x: 1}));
+      expect(await ctx.getConfig<number>('foo.a', 'x')).to.eql(1);
+      expect(await ctx.getConfig<number>('foo.a', 'y')).to.be.undefined();
+    });
+
+    it('defaults optional to true for config resolution', async () => {
+      // `servers.rest` does not exist yet
+      let server1port = await ctx.getConfig<number>('servers.rest', 'port');
+      expect(server1port).to.be.undefined();
+
+      // Now add `servers.rest`
+      ctx.configure('servers.rest').to({port: 3000});
+      server1port = await ctx.getConfig<number>('servers.rest', 'port');
+      expect(server1port).to.eql(3000);
+    });
+
+    it('throws error if a required config cannot be resolved', async () => {
+      await expect(
+        ctx.getConfig('servers.rest', 'host', {
+          optional: false,
+        }),
+      ).to.be.rejectedWith(/The key 'servers\.rest\:\$config' is not bound/);
+    });
+  });
+
+  describe('getConfigSync()', () => {
+    it('gets config for a binding', () => {
+      ctx.configure('foo').to({x: 1});
+      expect(ctx.getConfigSync('foo')).to.eql({x: 1});
+    });
+
+    it('throws a helpful error when the config is async', () => {
+      ctx.configure('foo').toDynamicValue(() => Promise.resolve('bar'));
+      expect(() => ctx.getConfigSync('foo')).to.throw(
+        /Cannot get config\[\] for foo synchronously: the value is a promise/,
+      );
+    });
+  });
+
+  describe('configResolver', () => {
+    class MyConfigResolver implements ConfigurationResolver {
+      getConfigAsValueOrPromise<ConfigValueType>(
+        key: BindingAddress<unknown>,
+        configPath?: string,
+        resolutionOptions?: ResolutionOptions,
+      ): ValueOrPromise<ConfigValueType | undefined> {
+        return ((key + '.config') as unknown) as ConfigValueType;
```

Comment:
I find this confusing. We use `.$config` suffix for config binding keys. When I see `.config` here, I have to think hard whether it's a type (missing `$` character), or whether this is just a random string.

Let's do this:

```suggestion
        return (`dummy config for ${key}`as unknown) as ConfigValueType;
```